crates/mun_abi/src/type_lut.rs

```diff
@@ -1,36 +1,81 @@
-use crate::Guid;
-use itertools::izip;
 use std::ffi::CStr;
+use std::hash::Hash;
 use std::os::raw::c_char;
 use std::{ffi, fmt, slice, str};
 
-/// Represents a unique identifier for types. The runtime can use this to lookup the corresponding [`TypeInfo`].
-#[repr(C)]
+use itertools::izip;
+
+use crate::Guid;
+
+/// Represents a unique identifier for types. The runtime can use this to lookup the corresponding
+/// [`TypeInfo`]. A [`TypeId`] is a key for a [`TypeInfo`].
+///
+/// A [`TypeId`] only contains enough information to query the runtime for a concrete type.
```

Comment:
> for a concrete type.

Are we mixing up terminology here? Can't the runtime contain a `*const *const i32`? This is not a concrete type, right?